addons/stock/tests/test_move.py

```diff
@@ -3767,4 +3767,27 @@ def test_put_in_pack_1(self):
         picking.button_validate()
         self.assertEqual(self.env['stock.quant']._get_available_quantity(self.product1, self.stock_location), 0)
         self.assertEqual(self.env['stock.quant']._get_available_quantity(self.product1, self.customer_location), 2)
+        
+    def test_change_tracking_1(self):
+        """changing the product's tracking method with reserved moves is
+        forbidden.
+        """
+        self.product1 = self.env.ref('stock.test_quant_product')
+
+        # Set a reserve quantity
+        move1 = self.env['stock.move'].create({
+            'name': 'test_in_1',
+            'location_id': self.supplier_location.id,
+            'location_dest_id': self.stock_location.id,
+            'product_id': self.product1.id,
+            'product_uom': self.uom_unit.id,
+            'product_uom_qty': 5.0,
+            'picking_type_id': self.env.ref('stock.picking_type_in').id,
+        })
+        move1._action_confirm()
+        move1._action_assign()
+
+        # Change the tracking method
+        with self.assertRaises(UserError):
+            self.product1.write({'tracking': 'lot'})
```

Comment:
won't raise once we integrate should bypass in the condition
make na internal move and tick scrap inseaft